### PR TITLE
Add GitLab storage block docs to deployment CLI

### PR DESCRIPTION
**Pull request by anna-geller**
Just adding docs

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: docs/concepts/storage.md [hunk 1]
```diff
@@ -44,13 +44,14 @@ Current options for deployment storage blocks include:
...
-| [Remote File System](/api-ref/prefect/filesystems/#prefect.filesystems.RemoteFileSystem) | Store data in a any filesystem supported by [`fsspec`](https://filesystem-spec.readthedocs.io/en/latest/). | |
-| [AWS S3 Storage](/api-ref/prefect/filesystems/#prefect.filesystems.S3) | Store data in an AWS S3 bucket. | [`s3fs`](https://s3fs.readthedocs.io/en/latest/) |
-| [Azure Storage](/api-ref/prefect/filesystems/#prefect.filesystems.Azure) | Store data in Azure Datalake and Azure Blob Storage. | [`adlfs`](https://github.com/fsspec/adlfs) |
-| [GitHub Storage](/api-ref/prefect/filesystems/#prefect.filesystems.GitHub) | Store data in a GitHub repository. | |
-| [Google Cloud Storage](/api-ref/prefect/filesystems/#prefect.filesystems.GCS) | Store data in a Google Cloud Platform (GCP) Cloud Storage bucket. | [`gcsfs`](https://gcsfs.readthedocs.io/en/latest/) |
-| [SMB](/api-ref/prefect/filesystems/#prefect.filesystems.SMB) | Store data in SMB shared network storage. | [`smbprotocol`](https://github.com/jborean93/smbprotocol) |
+| [Local File System](/api-ref/prefect/filesystems/#prefect.filesystems.LocalFileSystem) | Store code in a run's local file system. | |
```
Comment (anna-geller) [inline docs/concepts/storage.md:47]: FYI Terry: All I did here was change "Store data" to "Store code" + added GitLab

**Review by tpdorsey — APPROVED** [review]
Great add. Thank you

**Comment by anna-geller** [comment]
@tpdorsey, I added a bit more info - can you review it again? I realized the filesystem section also needs a reference. Thanks a lot! 🙌

**Comment by tpdorsey** [comment]
@anna-geller looks good. My approval stands. Ty

**Pull request merged**